src/main/java/yevhenii/lostfilmdemo/services/impl/TVSeriesServiceImpl.java

```diff
@@ -1,24 +1,84 @@
 package yevhenii.lostfilmdemo.services.impl;
 
+import lombok.RequiredArgsConstructor;
+import org.jooq.Record;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 import yevhenii.lostfilmdemo.entity.TVSeries;
+import yevhenii.lostfilmdemo.jooq.generated.tables.records.TvSeriesRecord;
+import yevhenii.lostfilmdemo.repository.JooqRepository;
 import yevhenii.lostfilmdemo.services.TVSeriesService;
 
+import java.util.List;
 import java.util.Optional;
+import java.util.stream.Collectors;
 
 @Service
+@RequiredArgsConstructor
 class TVSeriesServiceImpl implements TVSeriesService {
 
+    private final JooqRepository<String> schedulerRepository;
+
+    @Override
     @Transactional
+    public void save(TVSeries tvSeries) {
+
+        if (isExist(tvSeries.getLink()))
```

Comment:
instead of using this boolean check you can do it a bit nicer

`schedulerRepository.read(link).map(record -> this.update(record, tvSeries)).orElse(() -> this.save(new Record(), tvSeries));
 

`